core/src/main/java/org/testcontainers/containers/GenericContainer.java

```diff
@@ -135,7 +135,7 @@
     @NonNull
     private String networkMode;
 
-    @NonNull
+    @Nullable
```

Comment:
the annotation was wrong (we do allow resetting the network to `null` (no network))